docs/tutorials/using_apis/mesh/thread_intro.md

```diff
@@ -132,12 +136,72 @@ Now, the Thread network is ready to accept new joiner devices.
 
+#### Setting up Thread network
+
+##### Initial configuration
+
+This section gives some tips on how to initially configure the Thread network.
+
+You can create the initial static configuration for the Thread device can be done in the `.json` file. The following steps are what you need to do for the minimum setup.
+
+Usually, the border router is the first device to power up. 
+
+For the Thread border router, you need to configure the PSKc so that the commissioner can connect to it and start commissioning Thread devices to the network and modify the default network settings.
+
+For example, each border router could use the following practise for the initial settings (using `.json` format):
+
+- `"thread-config-panid": "unique random value",`
+- `"thread-config-network-name: "Some name",`
+- `"thread-config-extended-panid": "{unique random value}",`
+- `"thread-master-key: "{unique random value}",`
+- `"thread-config-pskc": "{generated value}",`
+
+The minimal configuration for the border router is to have input values for the PSKc generation, that is, the network name, extended panid and PSKc and the secret password, which is not stored to the device memory. Note that the only use for the extended panid is input for the PSKc generation in the Mbed Thread stack.
+
+You can use the panid to identify each network, but the only way to ensure that each Thread network stays in its own partition is to have different master key for each network.
+
+The Thread network channel may be different for each network to avoid unnecessary packet processing (if the networks can hear each other).
+
+- `"thread-config-channel": 22,`
+
+For the devices to be commissioned, the only thing needed is PSKd and eui64. You can set both values by using `ThreadInterface.h`: `device_eui64_set(const uint8_t *eui64)` and `mesh_error_t device_pskd_set(const char *pskd)`.
+
+Alternatively, you can configure PSKd in .json:
+
+`"thread-pskd": "Some random value",`
+
+By default, the radio driver reads the eui64.
+
+##### Device memory configuration
+
+Thread router devices require more RAM and ROM memory than end devices. How much RAM memory the device needs depends on the size of the network and how big and many packets need to be buffered.
+
+The recommended heap values to start with:
+
+- Thread router: 30kb (if using Arm Mbed Cloud Client and SEDs, then you might need more 2-5kb).
+- Thread SED/MED: 16kb.
+- Thread FED: 22kb.
+
+Also, it is possible to define the number of packets that can be buffered in the parent device. 
+
+If Thread end nodes are running Mbed Cloud Client, that sets requirements for the parent device to buffer the messages during the handshake. By default, the router device has 10 message buffer size to support the Mbed Cloud Client requirements. Note that if many end nodes start the Mbed Cloud Client at the same time, it may lead to the packet drops in the parent device.
```

Comment:
Query: Does it make sense to discuss Mbed Cloud Client in this part of the Mbed OS documentation?

Reply:
They are already aware of this.